### PR TITLE
Refactor Stashed Blob Upload Tracking

### PR DESCRIPTION
**Pull request by anthony-murphy**
When there are stash blob uploads the containerRuntime is blocked from connecting util those blobs are sucessfully uploaded, as we need to block replaying, and sending any ops that reference blob content, which hasn't been persisted to the server. The existing code managed to accomplish this, but the way it did it was not straightforward.  

One problem the existing code had is that it wasn't clear on what it meant to be a stashed blob, specifically if that meant the blob hadn't been uploaded, that the op for the blob hadn't been sent, and/or the op for the blob hadn't been acked. With this changed the stash state has been explicitly tied to the upload, as only the upload must complete before we can unblock the connection, as the ops for the blob will be handled naturally by applying the stashed ops, and replying the pending state, just like any other op. Specifically, the resubmit function in the blob manager will update the blob with the storage id which was obtained from the uploaded blob.

To further simplify the handling in the blob manager i've also removed the "uploading" property from the pending blobs, as all blobs in the pending map without a storage id are uploading.

I've also moved the tracking of blob uploads to trigger in the constuctor, to ensure it only runs once. The previous code checked that `this.delayConnectClientId === clientId` to simulate this run one behavior, but resulted in spawning as many promises as connections attempts, and make the code hard to follow.

Lastly, i've refactored the constructor of blob manager to use a props object for easier testing, and reduced some redundant code. 

fixed [AB#7617](https://dev.azure.com/fluidframework/235294da-091d-4c29-84fc-cdfc3d90890b/_workitems/edit/7617)

### CODE REVIEW

File: packages/runtime/container-runtime/src/test/blobManager.stashed.spec.ts [hunk 1]
```diff
@@ -0,0 +1,77 @@
+import { strict as assert } from "assert";
```
Comment (anthony-murphy) [inline packages/runtime/container-runtime/src/test/blobManager.stashed.spec.ts:1]: these tests are still in progress

**Comment by dannimad** [comment]
Are there any changes in behavior besides moving the tracking blob uploading to the constructor?  I see refactoring and simplifications, but maybe there are other subtle changes in the middle.

**Comment by anthony-murphy** [comment]
> Are there any changes in behavior besides moving the tracking blob uploading to the constructor? I see refactoring and simplifications, but maybe there are other subtle changes in the middle.

I call out all the expected changes in the description, its possible there are changes i'm unaware of, do you see any?

**Comment by dannimad** [comment]
No, I don't see any. I believe the only one that changes the behavior of previous code is the one I mentioned. I just wanted to understand the impact on functionality of this PR.

**Review by dannimad — APPROVED** [review]
LGTM
